### PR TITLE
Cancel animations that affect nodes that do not participate in layout.

### LINKED ISSUE #22378: Animations for nodes that are not visible are never removed from the list of running animations
```html
<style>
@keyframes boo {
  0%   { opacity: 0; }
  100% { opacity: 1; }
}
#box { animation: boo 1s infinite; }
</style>
<div id="box">hi there!</div>
<script>setTimeout(function() { document.getElementById('box').remove(); }, 3000)</script>
```
If you look at the list of layout events, the animation continues triggering after the node is removed from the document.

### PR DESCRIPTION
**Pull request by jdm**
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #22378
- [x] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22389)

### CODE REVIEW

File: components/script/dom/webidls/Window.webidl [hunk 1]
```diff
@@ -173,3 +173,8 @@ partial interface Window {
    readonly attribute TestRunner testRunner;
    //readonly attribute EventSender eventSender;
 };
+
+partial interface Window {
+   [Pref="css.animations.testing.enabled"]
+   readonly attribute unsigned long runningAnimations;
```
Comment (emilio) [inline components/script/dom/webidls/Window.webidl]: runningAnimationCount?

File: components/layout/animation.rs [hunk 2]
```diff
@@ -160,17 +160,36 @@ pub fn update_animation_state<E>(
...
+/// valid.
 pub fn recalc_style_for_animations<E>(
     context: &LayoutContext,
     flow: &mut dyn Flow,
     animations: &FxHashMap<OpaqueNode, Vec<Animation>>,
+) -> FxHashSet<OpaqueNode>
+where
+    E: TElement,
+{
+    let mut invalid_nodes = FxHashSet::default();
```
Comment (emilio) [inline components/layout/animation.rs]: `animations.keys().collect()`?

**Review by ferjm — APPROVED** [review]
LGTM with @emilio's comments addressed. Thanks!

**Comment by ferjm** [comment]
We need to make tidy happy before merging as well.

**Pull request merged**